src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc

```diff
@@ -529,6 +529,7 @@ static tsi_result handshaker_next(
     if (error != nullptr) *error = "invalid argument";
     return TSI_INVALID_ARGUMENT;
   }
+  if (received_bytes_size == 0) return TSI_INCOMPLETE_DATA;
```

Comment:
We have to be a bit careful: on the client-side, the first call to this API will have empty received bytes (and this is expected). To get around this, we can move this check just before the `if (handshaker->channel == nullptr ...)` line and change it to e.g.

`if (!handshaker->is_client && received_bytes_size == 0) { return TSI_INCOMPLETE_DATA; }`